sidecar/resources.py

```diff
@@ -154,6 +155,9 @@ def _watch_resource_loop(mode, *args):
                 raise
         except ProtocolError as e:
             print(f"{timestamp()} ProtocolError when calling kubernetes: {e}\n")
+        except MaxRetryError as e:
+            print(f"{timestamp()} MaxRetryError when calling kubernetes: {e}\n")
```

Comment:
Could you please enhance the log message to indicate that the process will wait a few seconds?

Reply:
Actually, I moved the sleep just before entering _watch_resource_iterator. This way, it catchs all exceptions.

I had the issue where the pods didn't have read secrets permission and it filled logs with millions of exceptions.